warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/edge/ProtobufEdgeDataTypeHandler.java

```diff
@@ -421,141 +169,44 @@ public void funnel(Key from, PrimitiveSink into) {
-        if (disallowlistFieldLookup.containsKey(dataType)) {
-            return this.disallowlistFieldLookup.get(dataType).contains(fieldName);
-        } else if (disallowlistFieldLookup.containsKey(EDGE_DEFAULT_DATA_TYPE)) {
-            // perhaps there is no disallowlist, which is fine
-            return this.disallowlistFieldLookup.get(EDGE_DEFAULT_DATA_TYPE).contains(fieldName);
-        }
-        return false;
-    }
-
-    private boolean isdisallowlistValue(String dataType, String fieldValue) {
-        if (disallowlistValueLookup.containsKey(dataType)) {
-            return this.disallowlistValueLookup.get(dataType).contains(fieldValue);
-        } else if (disallowlistValueLookup.containsKey(EDGE_DEFAULT_DATA_TYPE)) {
-            // perhaps there is no disallowlist, which is fine
-            return this.disallowlistValueLookup.get(EDGE_DEFAULT_DATA_TYPE).contains(fieldValue);
-        }
-        return false;
-    }
-
-    // used so we don't write duplicate stats entries for events with multiple field values;
-    protected Set<HashCode> activityLog = null;
-    protected Set<HashCode> durationLog = null;
-
-    protected BloomFilter<Key> activityLogBloom = null;
-    protected BloomFilter<Key> durationLogBloom = null;
-
     @Override
     public long process(KEYIN key, RawRecordContainer event, Multimap<String,NormalizedContentInterface> fields,
                     TaskInputOutputContext<KEYIN,? extends RawRecordContainer,KEYOUT,VALUEOUT> context, ContextWriter<KEYOUT,VALUEOUT> contextWriter)
                     throws IOException, InterruptedException {
-        long edgesCreated = 0;
-        long activityDate = -1;
-        boolean validActivityDate = false;
-        boolean activityEqualsEvent = false;
-        String edgeAttribute2 = null, edgeAttribute3 = null;
-
-        String loadDateStr = null;
 
-        if (event.fatalError()) {
-            return edgesCreated;
-        } // early short circuit return
-
-        // get edge definitions for this event type
+        long edgesCreated = 0;
         String typeName = getTypeName(event.getDataType());
 
-        List<EdgeDefinition> edgeDefs = null;
-        EdgeDefinitionConfigurationHelper edgeDefConfigs = null;
-        if (!edges.containsKey(typeName)) {
-            return edgesCreated; // short circuit, no edges defined for this type
+        // get edge definitions for this event type
+        edgeDefConfigs = edgeConfig.getEdges().get(typeName);
+        List<EdgeDefinition> edgeDefs = edgeDefConfigs.getEdges();
 
+        if (shortCircuit(event, typeName, edgeConfig.getEdges())) {
+            return edgesCreated;
         }
-        edgeDefConfigs = edges.get(typeName);
-        edgeDefs = edgeDefConfigs.getEdges();
 
-        trimFieldGroup = context.getConfiguration().getBoolean(typeName + TRIM_FIELD_GROUP, false);
+        // Track metadata for this event
+        eventMetadataRegistry = new HashMap<>();
```

Comment:
I suggest initializing this at its declaration and clearing the Map each time process is called, to avoid having to create a new HashMap in each call to process.